framework/web/Request.php

```diff
@@ -817,6 +854,82 @@ public function getQueryParam($name, $defaultValue = null)
         return isset($params[$name]) ? $params[$name] : $defaultValue;
     }
 
+    /**
+     * Sets the data related to the incoming request environment.
+     * @param array $serverParams server parameters.
+     * @since 2.1.0
+     */
+    public function setServerParams(array $serverParams)
+    {
+        $this->_serverParams = $serverParams;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @since 2.1.0
+     */
+    public function getServerParams()
+    {
+        if ($this->_serverParams === null) {
+            $this->_serverParams = $_SERVER;
+        }
+        return $this->_serverParams;
+    }
+
+    /**
+     * Return the server environment parameter by name.
+     * @param string $name parameter name.
+     * @param mixed $default default value to return if the parameter does not exist.
+     * @return mixed parameter value.
+     * @since 2.1.0
+     */
+    public function getServerParam($name, $default = null)
+    {
+        $params = $this->getServerParams();
+        if (!isset($params[$name])) {
+            return $default;
+        }
+        return $params[$name];
+    }
+
+    /**
+     * Specifies cookies.
+     * @param array $cookies array of key/value pairs representing cookies.
+     * @since 2.1.0
+     */
+    public function setCookieParams(array $cookies)
```

Comment:
Do we need it public?

Reply:
See https://github.com/yiisoft/yii2/pull/15416#discussion_r158818756